trio/_path.py

```diff
@@ -182,6 +182,70 @@ async def open(self, *args, **kwargs):
         value = await trio.to_thread.run_sync(func)
         return trio.wrap_file(value)
 
+    if TYPE_CHECKING:
```

Comment:
We don't need this `if TYPE_CHECKING` FYI, as the metaclass will overwrite these methods.

Probably OK though.

Reply:
It does make it more clear that these are for checkers only, and also improves import performance a tiny bit.